### PR TITLE
Allow Line.points definition to be a mix of lists/tuples

### PR DESCRIPTION
**Pull request by udiboy1209**
Fix for issue #3481

`Line.points` can be defined as list of lists, tuples and coordinates (float values).
Setter method modified to check each element of the argument list and append it to `self._points` accordingly.
Line example file /examples/canvas/line.py updated with the new possible way of defining points.

**Note:** appending a tuple to `points` causes problems

### CODE REVIEW

File: kivy/graphics/vertex_instructions_line.pxi [hunk 1]
```diff
@@ -672,8 +670,15 @@ cdef class Line(VertexInstruction):
         '''
         def __get__(self):
             return self._points
+
         def __set__(self, points):
-            self._points = list(points)
+            tmp = []
+            for p in points:
```
Comment (dessant) [inline kivy/graphics/vertex_instructions_line.pxi]: I haven't tested, but it looks like a slow implementation. I think mixing lists/tuples with plain values is not a priority. `points` could be either a plain list, a list of lists, or a list of tuples. This way, a faster algo could be used, something like what @kived proposed in the issue.

**Comment by udiboy1209** [comment]
@dessant I made the changes you requested. I agree the previous algorithm was slower. Given the conditions you mentioned we can use kived's algorithm

File: kivy/graphics/vertex_instructions_line.pxi [hunk 2]
```diff
@@ -672,8 +670,12 @@ cdef class Line(VertexInstruction):
         '''
         def __get__(self):
             return self._points
+
         def __set__(self, points):
-            self._points = list(points)
+            import itertools
```
Comment (dessant) [inline kivy/graphics/vertex_instructions_line.pxi]: Is there a reason for doing the import in the method?

File: kivy/graphics/vertex_instructions_line.pxi [hunk 3]
```diff
@@ -672,8 +670,12 @@ cdef class Line(VertexInstruction):
...
+            if points and isinstance(points[0], (list, tuple)):
```
Comment (dessant) [inline kivy/graphics/vertex_instructions_line.pxi]: What happens if points in not a list or tuple? :stuck_out_tongue_winking_eye:

File: kivy/graphics/vertex_instructions_line.pxi (hunk 2 above)
Comment (udiboy1209) [inline kivy/graphics/vertex_instructions_line.pxi]: I just realised that will slow things down :P

File: kivy/graphics/vertex_instructions_line.pxi (hunk 3 above)
Comment (udiboy1209) [inline kivy/graphics/vertex_instructions_line.pxi]: Oops! missed that!

**Comment by dessant** [comment]
:+1:

**Pull request merged**